src/posix/platform/system.cpp

```diff
@@ -175,11 +162,43 @@ void platformInit(otPlatformConfig *aPlatformConfig)
     ot::Posix::Udp::Get().Init(aPlatformConfig->mInterfaceName);
 #endif
 #endif
-
 exit:
     return;
 }
 
+void _platformInitOffloadMode(otPlatformConfig *aPlatformConfig)
+{
+    // Do nothing now.
+    OT_UNUSED_VARIABLE(aPlatformConfig);
+}
+
+void platformInit(otPlatformConfig *aPlatformConfig)
+{
+#if OPENTHREAD_POSIX_CONFIG_BACKTRACE_ENABLE
+    platformBacktraceInit();
+#endif
+
+    platformAlarmInit(aPlatformConfig->mSpeedUpFactor, aPlatformConfig->mRealTimeSignal);
+
+    sCoprocessorType = platformSpinelManagerInit(get802154RadioUrl(aPlatformConfig));
+
+    if (sCoprocessorType == OT_COPROCESSOR_RCP)
+    {
+        _platformInitCoreMode(aPlatformConfig);
+    }
+    else if (sCoprocessorType != OT_COPROCESSOR_NCP)
```

Comment:
```suggestion
    else if (sCoprocessorType == OT_COPROCESSOR_NCP)
```
May consider making it a `switch` clause.